### PR TITLE
[AMORO-2568] Add the spark container as another native option

### LINKED ISSUE #2568: [Subtask]: Add the spark container as another native option
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Parent issue

#2567

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

Close #2568 .

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by baiyangtx — CHANGES_REQUESTED** [review]
The spark container is a full feature, so I'd suggest that this PR also include the following：
1. Add configuration about SparkContainer in `docs/admin-guides/manager-optimizers`.  All supported properties should be listed.
2. Run the local test and take a screenshot of the test results in the PR's comments.
3. Add the configuration of the Spark Contianer as a comment in the default configuration file `dist/.../conf/config.yaml`

**Comment by tcodehuber** [comment]
> The spark container is a full feature, so I'd suggest that this PR also include the following：
> 
> 1. Add configuration about SparkContainer in `docs/admin-guides/manager-optimizers`.  All supported properties should be listed.
> 2. Run the local test and take a screenshot of the test results in the PR's comments.
> 3. Add the configuration of the Spark Contianer as a comment in the default configuration file `dist/.../conf/config.yaml`

OK, I will complete the todo list late.

**Comment by tcodehuber** [comment]
screenshot

**Comment by tcodehuber** [comment]
@baiyangtx Thanks a lot for your comments. I have tested in my local env, PTAL.

File: docker/build.sh [hunk 1]
```diff
@@ -25,11 +25,14 @@ cd $CURRENT_DIR
...
 FLINK_VERSION=1.15.3
+SPARK_VERSION=3.3.3
 HADOOP_VERSION=2.10.2
 DEBIAN_MIRROR=http://deb.debian.org
 APACHE_ARCHIVE=https://archive.apache.org/dist
-OPTIMIZER_JOB_PATH=ams/optimizer/flink-optimizer/target/flink-optimizer-${AMORO_VERSION}-jar-with-dependencies.jar
-OPTIMIZER_JOB=${PROJECT_HOME}/${OPTIMIZER_JOB_PATH}
+FLINK_OPTIMIZER_JOB_PATH=ams/optimizer/flink-optimizer/target/flink-optimizer-${AMORO_VERSION}-jar-with-dependencies.jar
+FLINK_OPTIMIZER_JOB=${PROJECT_HOME}/${FLINK_OPTIMIZER_JOB_PATH}
+SPARK_OPTIMIZER_JOB_PATH=ams/optimizer/spark-optimizer/target/spark-optimizer-${AMORO_VERSION}.jar
```
Comment (baiyangtx) [inline docker/build.sh:34]: The SPARK_OPTIMIZER_JOB_PATH should be a fatjar.

**Comment by tcodehuber** [comment]
> > @baiyangtx Thanks a lot for your comments. I have tested in my local env, PTAL.
> 
> Could you add some screenshot?

screenshot-20240308-182317
screenshot-20240308-182357
screenshot-20240308-182412

**Comment by tcodehuber** [comment]
> LGTM，plz fix the conflict. I think this PR is already to be merged.

Fixed.

File: ams/server/src/main/java/com/netease/arctic/server/manager/SparkOptimizerContainer.java [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+        Matcher matcher = APPLICATION_ID_PATTERN.matcher(readLine);
+        if (matcher.matches()) {
+          return String.format("application_%s_%s", matcher.group(2), matcher.group(3));
+        }
+        return null;
+      };
+  private String sparkMaster;
+  private DeployMode deployMode;
+  private String sparkHome;
+  private String sparkConfDir;
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/manager/SparkOptimizerContainer.java:72]: It seems to be not used. Do we still need it?

File: ams/server/src/main/java/com/netease/arctic/server/manager/SparkOptimizerContainer.java [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+        jobUri,
+        jobArgs);
+  }
+
+  private Map<String, String> loadSparkConfig() {
+    try {
+      return Arrays.stream(new org.apache.spark.SparkConf().getAll())
+          .collect(Collectors.toMap(Tuple2::_1, Tuple2::_2));
+    } catch (Exception e) {
+      LOG.error("load spark conf failed: {}", e.getMessage());
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/manager/SparkOptimizerContainer.java:196]:
```suggestion
      LOG.error("Load spark conf failed.", e);
```

**Review by zhoujinsong — COMMENTED** [review]
@tcodehuber 

The codes look good to me, but I still left some suggestion, PTAL.

File: ams/server/src/main/java/com/netease/arctic/server/manager/SparkOptimizerContainer.java (hunk 3 above)
Comment (tcodehuber) [reply ams/server/src/main/java/com/netease/arctic/server/manager/SparkOptimizerContainer.java]: @zhoujinsong Thx for your comments. It's really looks good to me. Fixed

File: ams/dashboard/src/main/resources/static/index.html [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="./favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Amoro</title>
+    <script type="module" crossorigin src="./assets/index-Sdc3fUIh.js"></script>
+    <link rel="stylesheet" crossorigin href="./assets/index-pOEir05u.css">
+  </head>
```
Comment (baiyangtx) [inline ams/dashboard/src/main/resources/static/index.html:28]: Please revert useless changes.

**Review by baiyangtx — COMMENTED** [review]
In addition, I think we should remove the spark-optimizer-job from amoro image.

see https://github.com/NetEase/amoro/blob/master/docker/amoro/Dockerfile#L40

**Comment by tcodehuber** [comment]
> In addition, I think we should remove the spark-optimizer-job from amoro image.
> 
> see https://github.com/NetEase/amoro/blob/master/docker/amoro/Dockerfile#L40

Fixed, thx for your comments.

**Review by zhoujinsong — APPROVED** [review]
LGTM.
Thanks for the contribution!

**Review by baiyangtx — APPROVED** [review]
LGMT, thanks for your contributions

**Pull request merged**